quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliConfigEncryptIT.java

```diff
@@ -0,0 +1,199 @@
+                        The secret %s was encrypted to
+                        """.formatted(SECRET_3.secret))
+                .assertCommandOutputNotContains("with the generated encryption key");
+    }
+
+    @DisabledOnOs(OS.WINDOWS) // Keytool command would require adjustments on Windows
+    @Order(4)
+    @Test
+    public void testKeyStoreConfigSourceWithSecrets() {
+        // this tests "Create Keystore" section
+        // see https://quarkus.io/version/main/guides/config-secrets#create-a-keystore
+        // unit test will check that Quarkus app retrieves secret correctly
+        var propertiesKeystoreName = "properties";
+        var propertiesKeystorePwd = "properties-password";
+        var encKeystoreName = "key";
+        var encKeystorePassword = "key-password";
+        // use keystores (one for actual secrets, one for secret encryption key)
+        encryptBuilder.getConfigCommand().addToApplicationPropertiesFile(
+                "smallrye.config.source.keystore.\"properties\".path", propertiesKeystoreName,
+                "smallrye.config.source.keystore.\"properties\".password", propertiesKeystorePwd,
+                "smallrye.config.source.keystore.\"properties\".handler", "aes-gcm-nopadding",
+                "smallrye.config.source.keystore.\"key\".path", encKeystoreName,
+                "smallrye.config.source.keystore.\"key\".password", encKeystorePassword);
+
+        // generate keystores
+        var encKeyBase64Encoded = base64.format(QuarkusEncryptConfigCommandBuilder.generateEncryptionKey());
+        encryptBuilder
+                .encryptionKeyFormat(plain)
+                .encryptionKeyFormatOpt(EncryptionKeyFormatOpt.LONG)
+                .secret(SECRET_4.secret)
+                .encryptionKey(encryptionKey)
+                .executeCommand()
+                .secretConsumer(Assertions::assertNotNull)
+                .secretConsumer(secret -> encryptBuilder
+                        .createKeyStore(SECRET_4.propertyName, secret, propertiesKeystoreName, propertiesKeystorePwd)
+                        .createKeyStore(AES_GCM_NO_PADDING_HANDLER_ENC_KEY, encKeyBase64Encoded, encKeystoreName,
+                                encKeystorePassword))
+                .assertApplicationPropertiesDoesNotContain(SECRET_4.secret)
+                .assertApplicationPropertiesDoesNotContain(SECRET_4.propertyName)
+                .assertApplicationPropertiesDoesNotContain(encKeyBase64Encoded)
+                .assertCommandOutputContains("""
+                        The secret %s was encrypted to
+                        """.formatted(SECRET_4.secret));
+    }
+
+    @Order(5)
+    @Test
+    public void testWrongSecretKeyHandler() {
+        // add unknown secret handler so that unit test can assert this is not reported
+        encryptBuilder.getConfigCommand()
+                .addToApplicationPropertiesFile(UNKNOWN_SECRET_HANDLER_PROPERTY, "${unknown-secret-handler::hush-hush}");
```

Comment:
I miss an assertion in this test.

Reply:
this capability asserts that property has been added

Reply:
here it is https://github.com/quarkus-qe/quarkus-test-suite/blob/6513942134a16ee9e849a1f0278e666eaa7b416b/quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/config/surefire/EncryptPropertyTest.java#L65

Reply:
so first we assert that we added config with unknown handler, then we assert that Quarkus reports issue

Reply:
Thanks for the details, comment on line should refer directly to `EncryptPropertyTest#testWrongSecretKeyHandler` unit test then

Reply:
done